Add support to take screenshots on macOS

dehesselle opened the pull request:
This is a follow-up to https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2424

- Introduce `screencapture` to support screen captures on macOS.
  - Using `-T0` for `full` is actually a workaround (it says "delay 0 seconds") as this parameter cannot be set to `None`.
- Use `platform.system()` instead of `os.name` as we need a way to detect macOS ("Darwin").

dehesselle committed:
Add support for screencapture on macOS

Use platform.system() as os.name does not distinguish between posix
platforms.

introt reviewed (changes requested):
Looks good, thanks!

Please update Plugins:Insert Screenshot to include "macOS: screencapture". If you want to go the extra mile, you may also add `(X11)` to the Linux options as per #2424 they don't work on Wayland

dehesselle committed:
Add hint about X11 requirement on Linux

The currently implemented screenshot functionality on Linux does not
work on Wayland.
https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2424

dehesselle commented:
As long as it stays a separate commit I'm not opposed to letting it piggyback on this PR.

introt reviewed (approved):
Thanks, LGTM!

introt commented:
...after  #2431?
